File: go-backend/dto/group.go
package dto

type GroupMemberDto struct {
	NodeId   int64  `json:"nodeId" binding:"required"`
	MemberIp string `json:"memberIp"`
	Priority int    `json:"priority"`
}

type GroupDto struct {
	Name       string           `json:"name" binding:"required"`
	Type       string           `json:"type" binding:"required"`
	SwitchBack int              `json:"switchBack"`
	Members    []GroupMemberDto `json:"members"`
}

type GroupUpdateDto struct {
	ID         int64            `json:"id" binding:"required"`
	Name       string           `json:"name"`
	SwitchBack *int             `json:"switchBack"`
	Members    []GroupMemberDto `json:"members"`
}

File: nextjs-frontend/lib/api/group.ts
import { post } from './client';

export type GroupMember = {
  id?: number;
  nodeId: number;
  nodeName?: string;
  memberIp?: string;
  priority: number;
  online?: boolean;
};

export type NodeGroup = {
  id: number;
  name: string;
  type: string;
  switchBack: number;
  activeMemberId: number;
  lastSwitchTime: number;
  inx: number;
  members: GroupMember[];
};

export type GroupMemberPayload = {
  nodeId: number;
  memberIp?: string;
  priority: number;
};

export type GroupCreatePayload = {
  name: string;
  type: string;
  switchBack: number;
  members: GroupMemberPayload[];
};

export type GroupUpdatePayload = {
  id: number;
  name?: string;
  switchBack?: number;
  members?: GroupMemberPayload[];
};

export const getGroupList = () => post<NodeGroup[]>('/group/list');
export const createGroup = (data: GroupCreatePayload) => post('/group/create', data);
export const updateGroup = (data: GroupUpdatePayload) => post('/group/update', data);
export const deleteGroup = (id: number) => post('/group/delete', { id });
export const updateGroupOrder = (items: { id: number; inx: number }[]) => post('/group/update-order', items);
